packages/edit-post/src/components/block-settings-menu/plugin-block-settings-menu-group.js

```diff
@@ -12,13 +12,14 @@ import { withSelect } from '@wordpress/data';
 
 const { Fill: PluginBlockSettingsMenuGroup, Slot } = createSlotFill( 'PluginBlockSettingsMenuGroup' );
 
-const PluginBlockSettingsMenuGroupSlot = ( { fillProps, selectedBlocks } ) => {
-	if ( !! selectedBlocks ) {
+export const PluginBlockSettingsMenuGroupSlot = ( { fillProps, selectedBlocks } ) => {
+	const blockNames = map( filter( selectedBlocks, ( block ) => block !== null && !! block.name ), ( block ) => block.name );
```

Comment:
It seems like you are only fixing one place where the same bugs might occur. We should check why does `getBlocksByClientId` work this way in the first place.

Reply:
Yes, we need to fix the root cause. Given that we're about to do a final RC, in the meanwhile I didn't want the editor to break so I proposed a partial fix in https://github.com/WordPress/gutenberg/pull/11996 which introduced a major regression (block settings menu slot not rendering).

I'm investigating why that happens, but wanted to fix the regression first in case I don't find a fix for the root cause. Another option would be to revent 11996 so the block settings menu plugin slot is not broken in the meantime.

Reply:
I'm going to revert the original PR for now to have time to think through this.

Reply:
Reverted in eee99f1ed711542301f630598e3847d5e6f33f46